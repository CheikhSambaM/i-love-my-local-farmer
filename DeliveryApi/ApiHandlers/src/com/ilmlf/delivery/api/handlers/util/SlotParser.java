package com.ilmlf.delivery.api.handlers.util;

import com.ilmlf.delivery.api.handlers.Slot;
import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;
import java.util.stream.StreamSupport;
import org.json.JSONArray;
import org.json.JSONObject;

public class SlotParser {

  /**
   * Parses the POST body and creates a list of Slot objects from it.
   *
   * @param body the Json formatted body of the request
   * @param farmIdStr the farmId as a String
   *
   * @return the List of Slot objects
   */
  public List<Slot> parseAndCreateSlotList(String body, String farmIdStr) {
    JSONObject bodyJson = new JSONObject(body);
    JSONArray slots = bodyJson.getJSONArray("slots");
    Stream<Object> slotsStream = StreamSupport.stream(slots.spliterator(), false);

    List<Slot> slotList = slotsStream
        .map(slot -> (Slot) parseAndCreateSlot((JSONObject) slot, farmIdStr))
        .collect(Collectors.toList());

    return slotList;
  }

  /**
   * Parses Json data and appends farmId to create a Slot object. <br/>
   * If any errors/exceptions encountered, will throw a RuntimeException
   *
   * @param slotJson the Json format slot data
   * @param farmIdStr the farmId as a String
   *
   * @return the Slot object
   */
  public static Slot parseAndCreateSlot(JSONObject slotJson, String farmIdStr) {
    String fromStr = slotJson.getString("from");
    LocalDateTime slotFrom = Slot.getLocalDateTimeFromIso(fromStr);

    Slot slot = Slot.builder()
        .farmId(Integer.parseInt(farmIdStr))
        .from(slotFrom)
        .to(Slot.getLocalDateTimeFromIso(slotJson.getString("to")))
        .availDeliveries(slotJson.getInt("numDeliveries"))
        .bookedDeliveries(0)
        .deliveryDate(slotFrom.toLocalDate())
        .build();

    return slot;
  }
}
