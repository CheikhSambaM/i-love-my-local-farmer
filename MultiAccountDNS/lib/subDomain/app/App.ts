import { Construct } from 'constructs';
import {
  Duration,
  aws_certificatemanager as certificateManager,
  aws_cloudfront as cloudFront,
  aws_cloudfront_origins as origins,
  aws_s3 as s3,
  aws_s3_deployment as s3deploy,
  aws_apigateway as apigateway,
  aws_lambda as lambda,
  aws_route53 as route53,
  custom_resources as customResources,
  aws_route53_targets as route53Targets,
} from 'aws-cdk-lib';
import { ICertificate } from 'aws-cdk-lib/aws-certificatemanager';

export interface FrontendProps {
  frontendDomains: string[];
  frontendCertificate: certificateManager.ICertificate;
  apiDomain: string;
  apiCertificate: certificateManager.ICertificate;
}

/** App Construct responsible for creating the frontend & backend Api resources for a simple demo App */
export class App extends Construct {
  public cloudFrontDistro: cloudFront.Distribution;
  public api: apigateway.LambdaRestApi;

  constructor(scope: Construct, id: string, props: FrontendProps) {
    super(scope, id);
    const { frontendDomains, frontendCertificate, apiDomain, apiCertificate } = props;

    const frontend = this.createFrontend(frontendDomains, frontendCertificate);
    this.api = this.createBackend(apiDomain, apiCertificate);
    const config = { apiEndpoint: 'https://' + apiDomain };
    this.deployPublicAssets(frontend.assetsBucket, frontend.cloudFrontDistro, config);

    this.cloudFrontDistro = frontend.cloudFrontDistro;
  }

  createFrontend(domainNames: string[], domainCertificate: ICertificate) {
    const assetsBucket = new s3.Bucket(this, 'Assets');

    const cloudFrontDistro = new cloudFront.Distribution(this, 'Distribution', {
      defaultBehavior: {
        origin: new origins.S3Origin(assetsBucket),
        viewerProtocolPolicy: cloudFront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
      },
      domainNames,
      certificate: domainCertificate,
      errorResponses: [
        {
          ttl: Duration.seconds(0),
          httpStatus: 404,
          responseHttpStatus: 200,
          responsePagePath: '/',
        },
      ],
      defaultRootObject: 'index.html',
    });

    return { cloudFrontDistro, assetsBucket };
  }

  createBackend(apiDomain: string, apiCertificate: certificateManager.ICertificate) {
    const [, env] = apiDomain.split('.');
    const handler = new lambda.Function(this, 'Api', {
      runtime: lambda.Runtime.NODEJS_12_X,
      handler: 'index.handler',
      code: lambda.Code.fromInline(`exports.handler = async () => ({
        'body': '${env}',
        'headers': {
          "Access-Control-Allow-Origin": "*",
          "Access-Control-Allow-Methods": "GET"
        }
      });`),
    });

    const api = new apigateway.LambdaRestApi(this, 'API', {
      handler,
      domainName: {
        domainName: apiDomain,
        certificate: apiCertificate,
        basePath: 'env',
      },
    });

    return api;
  }

  /** deploy a config file that contains any backend details to be served to the frontend */
  deployPublicAssets(bucket: s3.Bucket, cloudFrontDistro: cloudFront.Distribution, config: object) {
    const deployment = new s3deploy.BucketDeployment(this, 'DeployAssets', {
      sources: [s3deploy.Source.asset('./lib/subDomain/app/assets')],
      destinationBucket: bucket,
      distribution: cloudFrontDistro,
    });

    const filename = 'config.json';
    const putConfig = new customResources.AwsCustomResource(this, 'PutConfig', {
      onUpdate: {
        service: 'S3',
        action: 'putObject',
        parameters: {
          Bucket: bucket.bucketName,
          Key: filename,
          Body: JSON.stringify(config),
        },
        physicalResourceId: customResources.PhysicalResourceId.of('NO_DELETE_REQUIRED'),
      },
      policy: customResources.AwsCustomResourcePolicy.fromSdkCalls({
        resources: customResources.AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
    });
    putConfig.node.addDependency(deployment);
  }
}
