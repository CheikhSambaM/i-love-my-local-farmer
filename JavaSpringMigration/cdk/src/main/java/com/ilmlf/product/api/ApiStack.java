/*
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
Licensed under the Apache License, Version 2.0 (the "License").
You may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

package com.ilmlf.product.api;

import java.util.Map;

import lombok.Data;
import software.amazon.awscdk.core.Annotations;
import software.amazon.awscdk.core.Construct;
import software.amazon.awscdk.core.Fn;
import software.amazon.awscdk.core.Stack;
import software.amazon.awscdk.core.StackProps;
import software.amazon.awscdk.services.ec2.Vpc;
import software.amazon.awscdk.services.ecr.assets.DockerImageAsset;
import software.amazon.awscdk.services.ecr.assets.DockerImageAssetProps;
import software.amazon.awscdk.services.ecs.Cluster;
import software.amazon.awscdk.services.ecs.ClusterProps;
import software.amazon.awscdk.services.ecs.ContainerImage;
import software.amazon.awscdk.services.ecs.patterns.ApplicationLoadBalancedFargateService;
import software.amazon.awscdk.services.ecs.patterns.ApplicationLoadBalancedFargateServiceProps;
import software.amazon.awscdk.services.ecs.patterns.ApplicationLoadBalancedTaskImageOptions;
import software.amazon.awscdk.services.secretsmanager.ISecret;
import software.amazon.awscdk.services.secretsmanager.Secret;

public class ApiStack extends Stack {

  @lombok.Builder
  @Data
  public static class ApiStackProps implements StackProps {
    private String jdbcEndpointUrl;
    private String jdbcUsername;
    private String jdbcSecretArn;
    private String dbName;
    private Vpc vpc;
  }

  public ApiStack(final Construct scope, final String id, final ApiStackProps props) throws Exception {
    super(scope, id, props);

    DockerImageAsset imageAsset = this.createContainerImage();

    Cluster cluster = new Cluster(this, "ProvmanCluster", ClusterProps.builder()
        .vpc(props.vpc)
        .clusterName("ProvmanCluster")
        .build());

    ISecret dbPasswordSecret = Secret.fromSecretCompleteArn(this, "DbPasswordSecret", Fn.importValue(props.jdbcSecretArn));

    ApplicationLoadBalancedFargateService fargateService = new ApplicationLoadBalancedFargateService(this, "FargateService",
        ApplicationLoadBalancedFargateServiceProps.builder()
            .cluster(cluster)
            .taskImageOptions(ApplicationLoadBalancedTaskImageOptions.builder()
                .image(ContainerImage.fromDockerImageAsset(imageAsset))
                .containerPort(8080)
                .enableLogging(true)
                .secrets(Map.of("DB_PASSWORD",
                    software.amazon.awscdk.services.ecs.Secret.fromSecretsManager(dbPasswordSecret)))
                .environment(Map.of(
                    "DB_USERNAME", props.jdbcUsername,
                    "ENDPOINT_URL", "jdbc:mysql://" + Fn.importValue(props.jdbcEndpointUrl) + "/" + props.dbName
                ))
                .build())
            .desiredCount(1)
            .publicLoadBalancer(true)
            .serviceName("FargateEcsService")
            .build());

  }

  /**
   * Try to bundle the package locally. CDK can use this method to build locally
   * (which is faster). If the build doesn't work, it will build within a Docker
   * image which should work regardless of local environment.
   *
   * Note that CDK expects this function to return either true or false based on
   * bundling result.
   *
   * @param outputPath
   * @return whether the bundling script was successfully executed
   * @throws Exception
   */
  private Boolean buildWar(String outputPath) throws Exception {
    try {
      ProcessBuilder pb = new ProcessBuilder("bash", "-c",
          "cd ../app && mvn package && cp target/provman.war " + outputPath).inheritIO();

      Process p = pb.start(); // Start the process.
      p.waitFor(); // Wait for the process to finish.

      if (p.exitValue() == 0) {
        System.out.println("Script executed successfully");
        return true;
      } else {
        System.out.println("Script executed failed");
        throw new Error("Script executed failed");
      }

    } catch (Exception e) {
      e.printStackTrace();
      Annotations.of(this).addError(e.getMessage());
      throw e;
    }
  }

  public DockerImageAsset createContainerImage() throws Exception {
    this.buildWar("../cdk/src/main/container/");

    DockerImageAsset imageAsset = new DockerImageAsset(this, "ProvmanImage",
        DockerImageAssetProps.builder().directory("./src/main/container").build());

    return imageAsset;
  }
}
