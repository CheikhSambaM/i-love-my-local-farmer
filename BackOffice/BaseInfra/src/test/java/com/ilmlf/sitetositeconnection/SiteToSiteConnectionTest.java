package com.ilmlf.sitetositeconnection;


import io.github.jsonSnapshot.SnapshotMatcher;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import software.amazon.awscdk.core.App;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import static io.github.jsonSnapshot.SnapshotMatcher.expect;

import java.io.IOException;
import java.util.Map;

import org.junit.jupiter.api.Test;
import software.amazon.awscdk.core.AppProps;


public class SiteToSiteConnectionTest {
    private final static ObjectMapper JSON =
        new ObjectMapper().configure(SerializationFeature.INDENT_OUTPUT, true);

    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start(Snapshot::asJsonString);
    }

    @AfterAll
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    public void testStack() throws IOException {
        App app = new App(AppProps.builder().context(Map.of(
                "customerGatewayDeviceIP", "34.244.24.124",
                "vpcCidr", "172.30.0.0/16",
                "onPremiseCidr", "172.16.0.0/16"
        )).build());
        SiteToSiteConnectionStack stack = new SiteToSiteConnectionStack(app, "test");

        // synthesize the stack to a CloudFormation template
        JsonNode actual = JSON.valueToTree(app.synth().getStackArtifact(stack.getArtifactId()).getTemplate());

        expect(actual).toMatchSnapshot();

    }
}

