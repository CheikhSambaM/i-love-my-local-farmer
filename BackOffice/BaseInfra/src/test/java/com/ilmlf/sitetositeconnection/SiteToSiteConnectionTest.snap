com.ilmlf.sitetositeconnection.SiteToSiteConnectionTest.testStack=[
  {
    "Resources": {
      "VPCB9E5F0B4": {
        "Type": "AWS::EC2::VPC",
        "Properties": {
          "CidrBlock": "172.30.0.0/16",
          "EnableDnsHostnames": true,
          "EnableDnsSupport": true,
          "InstanceTenancy": "default",
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC"
            }
          ]
        }
      },
      "VPCPublicSubnet1SubnetB4246D30": {
        "Type": "AWS::EC2::Subnet",
        "Properties": {
          "CidrBlock": "172.30.0.0/18",
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "AvailabilityZone": {
            "Fn::Select": [
              0,
              {
                "Fn::GetAZs": ""
              }
            ]
          },
          "MapPublicIpOnLaunch": true,
          "Tags": [
            {
              "Key": "aws-cdk:subnet-name",
              "Value": "Public"
            },
            {
              "Key": "aws-cdk:subnet-type",
              "Value": "Public"
            },
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet1"
            }
          ]
        }
      },
      "VPCPublicSubnet1RouteTableFEE4B781": {
        "Type": "AWS::EC2::RouteTable",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet1"
            }
          ]
        }
      },
      "VPCPublicSubnet1RouteTableAssociation0B0896DC": {
        "Type": "AWS::EC2::SubnetRouteTableAssociation",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPublicSubnet1RouteTableFEE4B781"
          },
          "SubnetId": {
            "Ref": "VPCPublicSubnet1SubnetB4246D30"
          }
        }
      },
      "VPCPublicSubnet1DefaultRoute91CEF279": {
        "Type": "AWS::EC2::Route",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPublicSubnet1RouteTableFEE4B781"
          },
          "DestinationCidrBlock": "0.0.0.0/0",
          "GatewayId": {
            "Ref": "VPCIGWB7E252D3"
          }
        },
        "DependsOn": [
          "VPCVPCGW99B986DC"
        ]
      },
      "VPCPublicSubnet1EIP6AD938E8": {
        "Type": "AWS::EC2::EIP",
        "Properties": {
          "Domain": "vpc",
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet1"
            }
          ]
        }
      },
      "VPCPublicSubnet1NATGatewayE0556630": {
        "Type": "AWS::EC2::NatGateway",
        "Properties": {
          "AllocationId": {
            "Fn::GetAtt": [
              "VPCPublicSubnet1EIP6AD938E8",
              "AllocationId"
            ]
          },
          "SubnetId": {
            "Ref": "VPCPublicSubnet1SubnetB4246D30"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet1"
            }
          ]
        }
      },
      "VPCPublicSubnet2Subnet74179F39": {
        "Type": "AWS::EC2::Subnet",
        "Properties": {
          "CidrBlock": "172.30.64.0/18",
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "AvailabilityZone": {
            "Fn::Select": [
              1,
              {
                "Fn::GetAZs": ""
              }
            ]
          },
          "MapPublicIpOnLaunch": true,
          "Tags": [
            {
              "Key": "aws-cdk:subnet-name",
              "Value": "Public"
            },
            {
              "Key": "aws-cdk:subnet-type",
              "Value": "Public"
            },
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet2"
            }
          ]
        }
      },
      "VPCPublicSubnet2RouteTable6F1A15F1": {
        "Type": "AWS::EC2::RouteTable",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet2"
            }
          ]
        }
      },
      "VPCPublicSubnet2RouteTableAssociation5A808732": {
        "Type": "AWS::EC2::SubnetRouteTableAssociation",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPublicSubnet2RouteTable6F1A15F1"
          },
          "SubnetId": {
            "Ref": "VPCPublicSubnet2Subnet74179F39"
          }
        }
      },
      "VPCPublicSubnet2DefaultRouteB7481BBA": {
        "Type": "AWS::EC2::Route",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPublicSubnet2RouteTable6F1A15F1"
          },
          "DestinationCidrBlock": "0.0.0.0/0",
          "GatewayId": {
            "Ref": "VPCIGWB7E252D3"
          }
        },
        "DependsOn": [
          "VPCVPCGW99B986DC"
        ]
      },
      "VPCPublicSubnet2EIP4947BC00": {
        "Type": "AWS::EC2::EIP",
        "Properties": {
          "Domain": "vpc",
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet2"
            }
          ]
        }
      },
      "VPCPublicSubnet2NATGateway3C070193": {
        "Type": "AWS::EC2::NatGateway",
        "Properties": {
          "AllocationId": {
            "Fn::GetAtt": [
              "VPCPublicSubnet2EIP4947BC00",
              "AllocationId"
            ]
          },
          "SubnetId": {
            "Ref": "VPCPublicSubnet2Subnet74179F39"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PublicSubnet2"
            }
          ]
        }
      },
      "VPCPrivateSubnet1Subnet8BCA10E0": {
        "Type": "AWS::EC2::Subnet",
        "Properties": {
          "CidrBlock": "172.30.128.0/18",
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "AvailabilityZone": {
            "Fn::Select": [
              0,
              {
                "Fn::GetAZs": ""
              }
            ]
          },
          "MapPublicIpOnLaunch": false,
          "Tags": [
            {
              "Key": "aws-cdk:subnet-name",
              "Value": "Private"
            },
            {
              "Key": "aws-cdk:subnet-type",
              "Value": "Private"
            },
            {
              "Key": "Name",
              "Value": "test/VPC/PrivateSubnet1"
            }
          ]
        }
      },
      "VPCPrivateSubnet1RouteTableBE8A6027": {
        "Type": "AWS::EC2::RouteTable",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PrivateSubnet1"
            }
          ]
        }
      },
      "VPCPrivateSubnet1RouteTableAssociation347902D1": {
        "Type": "AWS::EC2::SubnetRouteTableAssociation",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPrivateSubnet1RouteTableBE8A6027"
          },
          "SubnetId": {
            "Ref": "VPCPrivateSubnet1Subnet8BCA10E0"
          }
        }
      },
      "VPCPrivateSubnet1DefaultRouteAE1D6490": {
        "Type": "AWS::EC2::Route",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPrivateSubnet1RouteTableBE8A6027"
          },
          "DestinationCidrBlock": "0.0.0.0/0",
          "NatGatewayId": {
            "Ref": "VPCPublicSubnet1NATGatewayE0556630"
          }
        }
      },
      "VPCPrivateSubnet2SubnetCFCDAA7A": {
        "Type": "AWS::EC2::Subnet",
        "Properties": {
          "CidrBlock": "172.30.192.0/18",
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "AvailabilityZone": {
            "Fn::Select": [
              1,
              {
                "Fn::GetAZs": ""
              }
            ]
          },
          "MapPublicIpOnLaunch": false,
          "Tags": [
            {
              "Key": "aws-cdk:subnet-name",
              "Value": "Private"
            },
            {
              "Key": "aws-cdk:subnet-type",
              "Value": "Private"
            },
            {
              "Key": "Name",
              "Value": "test/VPC/PrivateSubnet2"
            }
          ]
        }
      },
      "VPCPrivateSubnet2RouteTable0A19E10E": {
        "Type": "AWS::EC2::RouteTable",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC/PrivateSubnet2"
            }
          ]
        }
      },
      "VPCPrivateSubnet2RouteTableAssociation0C73D413": {
        "Type": "AWS::EC2::SubnetRouteTableAssociation",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPrivateSubnet2RouteTable0A19E10E"
          },
          "SubnetId": {
            "Ref": "VPCPrivateSubnet2SubnetCFCDAA7A"
          }
        }
      },
      "VPCPrivateSubnet2DefaultRouteF4F5CFD2": {
        "Type": "AWS::EC2::Route",
        "Properties": {
          "RouteTableId": {
            "Ref": "VPCPrivateSubnet2RouteTable0A19E10E"
          },
          "DestinationCidrBlock": "0.0.0.0/0",
          "NatGatewayId": {
            "Ref": "VPCPublicSubnet2NATGateway3C070193"
          }
        }
      },
      "VPCIGWB7E252D3": {
        "Type": "AWS::EC2::InternetGateway",
        "Properties": {
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC"
            }
          ]
        }
      },
      "VPCVPCGW99B986DC": {
        "Type": "AWS::EC2::VPCGatewayAttachment",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "InternetGatewayId": {
            "Ref": "VPCIGWB7E252D3"
          }
        }
      },
      "VPCVpnGatewayB5ABAE68": {
        "Type": "AWS::EC2::VPNGateway",
        "Properties": {
          "Type": "ipsec.1",
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC"
            }
          ]
        }
      },
      "VPCVPCVPNGWEF1DEEBE": {
        "Type": "AWS::EC2::VPCGatewayAttachment",
        "Properties": {
          "VpcId": {
            "Ref": "VPCB9E5F0B4"
          },
          "VpnGatewayId": {
            "Ref": "VPCVpnGatewayB5ABAE68"
          }
        }
      },
      "VPCRoutePropagation4992C07E": {
        "Type": "AWS::EC2::VPNGatewayRoutePropagation",
        "Properties": {
          "RouteTableIds": [
            {
              "Ref": "VPCPrivateSubnet1RouteTableBE8A6027"
            },
            {
              "Ref": "VPCPrivateSubnet2RouteTable0A19E10E"
            }
          ],
          "VpnGatewayId": {
            "Ref": "VPCVpnGatewayB5ABAE68"
          }
        },
        "DependsOn": [
          "VPCVPCVPNGWEF1DEEBE"
        ]
      },
      "VPCstaticCustomerGatewayCC72CA46": {
        "Type": "AWS::EC2::CustomerGateway",
        "Properties": {
          "BgpAsn": 65000,
          "IpAddress": "34.244.24.124",
          "Type": "ipsec.1",
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC"
            }
          ]
        }
      },
      "VPCstatic2DBA99F3": {
        "Type": "AWS::EC2::VPNConnection",
        "Properties": {
          "CustomerGatewayId": {
            "Ref": "VPCstaticCustomerGatewayCC72CA46"
          },
          "Type": "ipsec.1",
          "StaticRoutesOnly": true,
          "Tags": [
            {
              "Key": "Name",
              "Value": "test/VPC"
            }
          ],
          "VpnGatewayId": {
            "Ref": "VPCVpnGatewayB5ABAE68"
          }
        }
      },
      "VPCstaticRoute1721600165A5FD45F": {
        "Type": "AWS::EC2::VPNConnectionRoute",
        "Properties": {
          "DestinationCidrBlock": "172.16.0.0/16",
          "VpnConnectionId": {
            "Ref": "VPCstatic2DBA99F3"
          }
        }
      }
    }
  }
]